### PR TITLE
Fix conf/ folder checks: privilege and resource are JSON

### PR DESCRIPTION
**Pull request by ymartin59**
Propagate Python exception to uwsgi log

### CODE REVIEW

**Comment by Diaoul** [comment]
Is this still something we need?

**Comment by ymartin59** [comment]
@Diaoul I would say yes, as it is still possible an invalid json file is included in uploaded package and break publish operation silently... having a log really helps.

**Comment by ymartin59** [comment]
With gunicorn 500 Internal Error and stack traces in spkrepo docker container logs confirm this fix is required

**Comment by Diaoul** [comment]
Did the previous code work at some point? If not then I guess there will be no backward compatibility issue..

**Comment by Diaoul** [comment]
I've looked at the database, there is currently no package with `conf_privilege` or `conf_resource`. I guess that answers my question.

File: spkrepo/utils.py [hunk 1]
```diff
@@ -185,27 +185,23 @@ def __init__(self, stream):
                             {s: {k: v for k, v in c.items(s)} for s in c.sections()}
                         )
                     if "conf/privilege" in names:
-                        c = ConfigParser()
                         try:
-                            c.read_string(
+                            self.conf_privilege = json.loads(
```
Comment (Diaoul) [inline spkrepo/utils.py:189]: This is wrong, `conf_privilege` should be a string not a dict.

File: spkrepo/utils.py [hunk 2]
```diff
@@ -185,27 +185,23 @@ def __init__(self, stream):
...
-                        self.conf_privilege = json.dumps(
-                            {s: {k: v for k, v in c.items(s)} for s in c.sections()}
-                        )
+                        except ValueError:
+                            raise SPKParseError("Invalid JSON conf/privilege")
                     if "conf/resource" in names:
-                        c = ConfigParser()
                         try:
-                            c.read_string(
+                            self.conf_resource = json.loads(
```
Comment (Diaoul) [inline spkrepo/utils.py:198]: This is wrong, `conf_resource` should be a string not a dict.

**Comment by ymartin59** [comment]
@Diaoul Correct. I have selected parts of Dr Bean DSM 6 branch and push them in production as patches "to get it work" - without DB updates to avoid troubles. So this validation code from my branch was in production since a while but neither `conf_resource` or `conf_privilege` was used later.
By the way, even if I am not a good Python developer (and I fear I will never be), I have to setup a proper dev environment and do things the right way... I come back soon when I have learned prerequisites.

**Comment by Diaoul** [comment]
No worries I will handle it, I just didn't want us both to work on the same issue! :cake:

**Comment by Diaoul** [comment]
Fixed with 071f9ee33ec3b398ecd8516cd1b1d6055e0f11ac

**Comment by Diaoul** [comment]
I've updated the server as well, please try to publish again.

**Pull request closed without merging**